Support mounting volumes for secrets/configmaps items

Sgitario opened the pull request:
Adding support of volume items only for secrets and configmap for Knative, OpenShift and Kubernetes deployments.

The new properties looks like:

```
quarkus.kubernetes.mounts.volume-certs.path=/etc/certs
quarkus.kubernetes.secret-volumes.volume-certs.secret-name=root-secret
quarkus.kubernetes.secret-volumes.volume-certs.items.keystore.path=keystore.p12
```

Where items is a map of string (the item key) and items (the item path + the item mode).

Fix https://github.com/quarkusio/quarkus/issues/25058

iocanel commented:
On it!

iocanel commented on extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VolumeItem.java:
```diff
@@ -0,0 +1,19 @@
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class VolumeItem {
```
I would name it `VolumeConfig` to align with the rest of the code.

Sgitario replied:
I guess you mean VolumeItemConfig, right ?

iocanel replied:
`VolumeItemConfig` or `VolumeConfig` both are fine.

iocanel reviewed (approved):
LGTM

Neat peaking: I would probably rename `VolumeItem` to `VolumeConfig` for consistency. But defintely not a blocker.

Sgitario committed:
Support mounting volumes for secrets/configmaps items

Adding support of volume items only for secrets and configmap for Knative, OpenShift and Kubernetes deployments.

The new properties looks like:

```
quarkus.kubernetes.mounts.volume-certs.path=/etc/certs
quarkus.kubernetes.secret-volumes.volume-certs.secret-name=root-secret
quarkus.kubernetes.secret-volumes.volume-certs.items.keystore.path=keystore.p12
```

Where items is a map of string (the item key) and items (the item path + the item mode).

Fix https://github.com/quarkusio/quarkus/issues/25058
